skuba-update/skuba_update/skuba_update.py

```diff
@@ -68,19 +88,39 @@ def main():
 
     # Check that we have the proper zypper version.
     if not check_version('zypper', REQUIRED_ZYPPER_VERSION):
-        raise Exception('zypper version {0} or higher is required'.format(
-            '.'.join([str(x) for x in REQUIRED_ZYPPER_VERSION])
-        ))
+        raise ZypperVersionTooLow(
+            'zypper version {0} or higher is required'.format(
+                '.'.join([str(x) for x in REQUIRED_ZYPPER_VERSION])
+            )
+        )
 
     if os.geteuid() != 0:
-        raise Exception('root privileges are required to run this tool')
+        raise PrivilegesTooLow('root privileges are required to run this tool')
 
     run_zypper_command(['ref', '-s'])
+
     if not args.annotate_only:
-        code = update()
+        pkgs = parse_kubelet_pkglist(get_kubelet_packages_details())
```

Comment:
should I just catch the case where no kubelet is installed, directly log() an error, and  fail?

Reply:
It is already failing if no kubelet if found right? I don't think there is need to be more specific or having an special way of handling this particular case. I just checked `skuba-update` package requires `kubernetes-client`, probably it would make sense to also add `kubernetes-kubelet` ass a requirement. However if this requirement is not matched I'd say this the node is terribly broken and I'd consider `skuba-update` failures mostly meaningless in that case.

Reply:
I am not sure if it's required to have kubelet as a requirement upon installing. Sensu stricto, skuba-update doesn't require kubelet. It just won't work until kubelet is installed. I am okay if someone wants to change the packaging though.